ZenPacks/zenoss/HttpMonitor/http.py

```diff
@@ -107,7 +134,8 @@ def _agent(self):
                 timeout=self._timeout
             )
             agent = ProxyAgent(endpoint)
-        agent = RedirectAgent(agent) if self._follow else agent
+        agent = RedirectAgentZ(agent, onRedirect=self._follow, port=self._port, proxy=self._proxyIp) \
+                if self._follow in ('follow', 'sticky', 'stickyport') else agent
```

Comment:
For readability, using a regular `if` statement is probably better here than an expression:
```
    if self._follow in ('follow', 'sticky', 'stickyport'):
        agent = RedirectAgentZ(
            agent, onRedirect=self._follow, port=self._port, proxy=self._proxyIp
        )
```